pkg/server/commands/listusers/list_users_rpc_test.go

```diff
@@ -2079,6 +2080,87 @@ func TestListUsersCycleDetection(t *testing.T) {
 	})
 }
 
+func TestListUsersDepthExceeded(t *testing.T) {
+	model := `model
+	schema 1.1
+		type user
+
+	type folder
+		relations
+			define parent: [folder]
+			define viewer: [user] or viewer from parent`
+
+	depthNotExceededTuples := []*openfgav1.TupleKey{
+		tuple.NewTupleKey("folder:26", "viewer", "user:maria"),
+		tuple.NewTupleKey("folder:25", "parent", "folder:26"),
+		tuple.NewTupleKey("folder:24", "parent", "folder:25"),
+		tuple.NewTupleKey("folder:23", "parent", "folder:24"),
+		tuple.NewTupleKey("folder:22", "parent", "folder:23"),
+		tuple.NewTupleKey("folder:21", "parent", "folder:22"),
+		tuple.NewTupleKey("folder:20", "parent", "folder:21"),
+		tuple.NewTupleKey("folder:19", "parent", "folder:20"),
+		tuple.NewTupleKey("folder:18", "parent", "folder:19"),
+		tuple.NewTupleKey("folder:17", "parent", "folder:18"),
+		tuple.NewTupleKey("folder:16", "parent", "folder:17"),
+		tuple.NewTupleKey("folder:15", "parent", "folder:16"),
+		tuple.NewTupleKey("folder:14", "parent", "folder:15"),
+		tuple.NewTupleKey("folder:13", "parent", "folder:14"),
+		tuple.NewTupleKey("folder:12", "parent", "folder:13"),
+		tuple.NewTupleKey("folder:11", "parent", "folder:12"),
+		tuple.NewTupleKey("folder:10", "parent", "folder:11"),
+		tuple.NewTupleKey("folder:9", "parent", "folder:10"),
+		tuple.NewTupleKey("folder:8", "parent", "folder:9"),
+		tuple.NewTupleKey("folder:7", "parent", "folder:8"),
+		tuple.NewTupleKey("folder:6", "parent", "folder:7"),
+		tuple.NewTupleKey("folder:5", "parent", "folder:6"),
+		tuple.NewTupleKey("folder:4", "parent", "folder:5"),
+		tuple.NewTupleKey("folder:3", "parent", "folder:4"),
+		tuple.NewTupleKey("folder:2", "parent", "folder:3"), // folder:2 will not exceed depth limit of 25
+		tuple.NewTupleKey("folder:1", "parent", "folder:2"), // folder:1 will exceed depth limit of 25
+	}
+
+	tests := ListUsersTests{
+		{
+			name: "depth_should_exceed_limit",
+			req: &openfgav1.ListUsersRequest{
+				Object: &openfgav1.Object{
+					Type: "folder",
+					Id:   "1", // Exceeded because we expand up until folder:1, beyond 25 allowable levels
+				},
+				Relation: "viewer",
+				UserFilters: []*openfgav1.ListUsersFilter{
+					{
+						Type: "user",
+					},
+				},
+			},
+			model:            model,
+			tuples:           depthNotExceededTuples,
+			expectedErrorMsg: "resolution depth exceeded",
```

Comment:
Can we please assert on the sentinel error instead of the error message? The reasoning is because the internal API contract is on the sentinel err `ErrResolutionDepthExceeded`, not the error message. The error message can change but the sentinel error itself won't.